testplan/common/utils/match.py

```diff
@@ -400,3 +413,63 @@ def get_between(self, mark1=None, mark2=None):
                 lines_between.append(line)
             separator = b"" if self.binary else ""
             return separator.join(lines_between)
+
+    @contextmanager
+    def expect(
+        self,
+        result: "Result",
+        regex: Union[List[Regex], Regex],
+        timeout: Union[List[float], float] = LOG_MATCHER_DEFAULT_TIMEOUT,
+        strict_order: bool = True,
+    ):
+        """
+        User-friendly context manager composed of ``seek_eof``, ``match`` and
+        Testplan assertions. On entering jumping to log stream EOF, on exiting
+        doing log matching as expected patterns should be (indirectly) produced
+        by lines of Python code in context manager body.
+
+        :param result: Testplan Result object used in assertions
+        :param regex: Regex string or compiled regular expression or a list of
+            such regex strings or expressions
+        :param timeout: Timeout in seconds for regex matching, can be a list
+            indicating different individual timeout, or be a float applied to
+            all input regexes.
+        :param strict_order: Of value True by default, indicating sequential
+            matching of input regexes. Set to False for supporting out of order
+            matching.
+        """
+        self.seek_eof()
+        s_pos = self.position
+        self.expect_results = []
+
+        if not isinstance(regex, list):
+            regex = [regex]
+        if not isinstance(timeout, list):
+            timeout = repeat(timeout, len(regex))
+        elif len(timeout) != len(regex):
+            raise ValueError(
+                "``timeout`` length doesn't match ``regex`` length."
+            )
+        pairs = zip(regex, timeout)
+
+        yield
+
+        for r, t in pairs:
+            m = self.match(r, t, raise_on_timeout=False)
```

Comment:
the first match might not run from s_pos if the log matcher was used within the context manager

Reply:
right... i need to think it over 👍